extensions/smallrye-graphql/deployment/src/main/java/io/quarkus/smallrye/graphql/deployment/SmallryeGraphqlProcessor.java

```diff
@@ -0,0 +1,384 @@
+            routeProducer.produce(new RouteBuildItem(quarkusConfig.rootPathUi + "/*", handler));
+            notFoundPageDisplayableEndpointProducer
+                    .produce(new NotFoundPageDisplayableEndpointBuildItem(quarkusConfig.rootPathUi + "/"));
+        } else if (quarkusConfig.alwaysIncludeUi) {
+            ResolvedArtifact artifact = getGraphQLUiArtifact();
+            //we are including in a production artifact
+            //just stick the files in the generated output
+            //we could do this for dev mode as well but then we need to extract them every time
+            File artifactFile = artifact.getArtifactPath().toFile();
+            try (JarFile jarFile = new JarFile(artifactFile)) {
+                Enumeration<JarEntry> entries = jarFile.entries();
+
+                while (entries.hasMoreElements()) {
+                    JarEntry entry = entries.nextElement();
+                    if (entry.getName().startsWith(GRAPHQL_UI_WEBJAR_PREFIX) && !entry.isDirectory()) {
+                        try (InputStream inputStream = jarFile.getInputStream(entry)) {
+                            String filename = entry.getName().replace(GRAPHQL_UI_WEBJAR_PREFIX + "/", "");
+                            byte[] content = FileUtil.readFileContents(inputStream);
+                            if (entry.getName().endsWith(FILE_TO_UPDATE)) {
+                                content = updateApiUrl(new String(content, StandardCharsets.UTF_8), graphQLPath)
+                                        .getBytes(StandardCharsets.UTF_8);
+                            }
+                            if (IGNORE_LIST.contains(filename)) {
+                                ClassLoader classLoader = SmallryeGraphqlProcessor.class.getClassLoader();
+                                InputStream resourceAsStream = classLoader
+                                        .getResourceAsStream(OWN_MEDIA_FOLDER + filename);
+                                content = streamToByte(resourceAsStream);
+                            }
+
+                            String fileName = GRAPHQL_UI_FINAL_DESTINATION + "/" + filename;
+
+                            generatedResourceProducer
+                                    .produce(new GeneratedResourceBuildItem(fileName, content));
+
+                            nativeImageResourceProducer
+                                    .produce(new NativeImageResourceBuildItem(fileName));
+
+                        }
+                    }
+                }
+            }
+
+            Handler<RoutingContext> handler = recorder
+                    .uiHandler(GRAPHQL_UI_FINAL_DESTINATION, httpRootPath.adjustPath(quarkusConfig.rootPathUi));
+            routeProducer.produce(new RouteBuildItem(quarkusConfig.rootPathUi, handler));
+            routeProducer.produce(new RouteBuildItem(quarkusConfig.rootPathUi + "/*", handler));
+        }
+    }
+
+    private byte[] streamToByte(InputStream is) throws IOException {
```

Comment:
Could we use `org.apache.commons.io.IOUtils.toByteArray(InputStream)` instead? It seems to be somewhere among the dependencies and some other extensions use it too

Reply:
Happy to use that, did not want to introduce an new dependency, but if it's there it makes sense. Let me know.

Reply:
Please check before doing that that the dependency is not there by pure luck (i.e. brought by another one that is not really related to what we do and where it could go away)

Reply:
Seems like it's available:
GroupId: commons-io 
ArtifactId: commons-io 
Version: 2.6 
Type: jar Scope: compile 

Via: maven-shared-utils 
Via: maven-embedder 
Via: quarkus-bootstrap-core 
Via: quarkus-core-deployment

Should I go ahead and use that ? @gsmet ?

Reply:
Given it's coming incidentally from Maven, I wouldn't use it.

Better add what you need to `io.quarkus.deployment.util.IoUtil`.

Reply:
No need to add anything, the method was there already. Doing a fix now